presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/DefaultThriftHeaderProvider.java

```diff
@@ -22,7 +22,7 @@
         implements ThriftHeaderProvider
 {
     @Override
-    public Map<String, String> getHeaders(ConnectorSession session)
+    public Map<String, String> getHeaders(ConnectorSession session, ThriftConnectorConfig config)
```

Comment:
Can you not inject ThriftConnectorConfig when constructing the instance?

Reply:
(Note: I am a complete Java n00b, so this might be totally wrong).

I don't think we can inject `ThriftConnectorConfig`.  There are potentially many configs, and these configs are loaded at runtime (corresponding to a particular Thrift Connector).